src/shogun/machine/IterativeMachine.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Shubham Shukla
+ */
+
+#ifndef _ITERATIVEMACHINE_H__
+#define _ITERATIVEMACHINE_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/SGVector.h>
+#include <shogun/lib/common.h>
+#include <shogun/machine/LinearMachine.h>
+
+namespace shogun
+{
+	class CFeatures;
+	class CLabels;
+
+#define SG_ADD8(param, name, description, ms_available)                        \
```

Comment:
Rather than defining this here, couldn't we just add this `this->` to the original SG_ADD?
@vigsterkr @lisitsyn

Reply:
Wow, what's 8 here? :)

I very much prefer getting rid of macro at all, if possible.

Reply:
the 8 is to diffrentiate between SG_ADD and the macro here... not anything meaningful really

Reply:
Any potential side-effects adding it? Maybe we can add it and run the tests, examples, etc, to get a feeling of it.

Reply:
acutually no. Just talked with @vigsterkr and we agreed that we should just add the "this" in the orginal definition of the macro